src/ray/raylet/worker_pool.h

```diff
@@ -538,10 +562,13 @@ class WorkerPool : public WorkerPoolInterface, public IOWorkerPoolInterface {
     /// the extra information of the process. Note that the shim process PID is the
     /// same with worker process PID, except worker process in container.
     absl::flat_hash_map<StartupToken, WorkerProcessInfo> worker_processes;
-    /// A map for looking up the task by the startup token of starting worker process.
-    absl::flat_hash_map<StartupToken, TaskWaitingForWorkerInfo> starting_workers_to_tasks;
-    /// Pop worker requests that are pending due to maximum_startup_concurrency_.
-    std::deque<PopWorkerRequest> pending_pop_worker_requests;
+    /// FIFO queue of pending requests with workers STARTED but pending registration.
+    /// If a request stays in this status for >= worker_register_timeout_seconds, we'll
+    /// fail the request and kill the worker process.
+    std::deque<std::shared_ptr<PopWorkerRequest>> pending_registration_requests;
+    /// FIFO queue of pending requests with workers NOT STARTED due to
+    /// maximum_startup_concurrency_.
+    std::deque<std::shared_ptr<PopWorkerRequest>> pending_start_requests;
```

Comment:
Could you include the queue size in the DebugString()

Reply:
done